amps/ags/rm-community/rm-community-rest-api-explorer/src/main/webapp/definitions/gs-core-api.yaml

```diff
@@ -2495,7 +2497,194 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  ##retention-schedule
+  '/record-categories/{recordCategoryId}/retention-schedules':
+    post:
+      tags:
+        - retention-schedule
+      summary: Create a retention schedule
+      description: |
+        Create a retention schedule.        
+        For example, using the following JSON body will create a retention schedule:
+        ```JSON
+          {
+              "authority":"Retention Authority",
+              "instructions":"Retention Instructions",
+              "isRecordLevel":false
+          }
+        ```
+      operationId: createRetentionSchedule
+      parameters:
+        - $ref: '#/parameters/recordCategoryIdParam'
+        - in: body
+          name: nodeBodyCreate
+          description: |
+            The node information to create.
+          schema:
+            $ref: '#/definitions/RetentionNodeBodyCreate'
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      responses:
+        '200':
+          description: Successful response
+          schema:
+            $ref: '#/definitions/RetentionScheduleResponse'
+        '400':
+          description: |
+            Invalid parameter: recordCategoryId is not a valid format
+        '401':
+          description: Authentication failed
+        '403':
+          description: Current user does not have permission to create retention schedule
+        '404':
+          description: recordCategoryId does not exist
+        default:
```

Comment:
We should mention the error that happens when the record category already has a retention schedule, and what happens if the category already has members, and so the schedule can't be created.

Reply:
Ah interesting. Maybe I misremembered and the schedule can't be changed at that point? I thought it was impossible to create a record-level schedule if there were already records in the category.

Reply:
I just got around to testing this and I was right. Is this behaviour different when you use the API?
![image](https://github.com/Alfresco/alfresco-community-repo/assets/11535082/29a8f2ac-f44d-4a95-af4e-fa04053b1069)

Reply:
Yes, API also behaving same when we have record in the category, we can't have a record-level schedule for it.